### PR TITLE
huggingface_hub 0.23.0

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4803](https://anaconda.atlassian.net/browse/PKG-4803)
- [Upstream repository](https://github.com/huggingface/huggingface_hub/tree/v0.23.0)
- [Upstream changelog/diff](https://github.com/huggingface/huggingface_hub/releases)
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/datasets-feedstock/pull/4

### Explanation of changes:

- Update version and dependencies

[PKG-4803]: https://anaconda.atlassian.net/browse/PKG-4803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,11 +32,26 @@ requirements:
     - tqdm >=4.42.1
     - typing-extensions >=3.7.4.3
   run_constrained:
+    # cli
+    - InquirerPy ==0.3.4
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - inquirerpy ==0.3.4
```
see https://pypi.org/project/inquirerpy/

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,11 +32,26 @@ requirements:
...
     - fastcore >=1.3.27
-    - InquirerPy ==0.3.4
-    - pydantic >1.1,<3.0  # [py>38]
-    - pydantic >1.1,<2.0  # [py==38]
+    # tensorflow
+    - tensorflow
+    - pydot
+    - graphviz
+    # tensorflow-testing
+    - keras <3.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: In `run_constrained` there should be packages **with** pinnings only.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 {% set name = "huggingface_hub" %}
-{% set version = "0.20.3" %}
+{% set version = "0.23.0" %}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
{% set version = "0.23.1" %}
```
They just released a hotfix https://github.com/huggingface/huggingface_hub/releases/tag/v0.23.1
https://github.com/huggingface/huggingface_hub/pull/2271
Mentioned as solving bugs in datasets.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -32,11 +32,17 @@ requirements:
     - tqdm >=4.42.1
     - typing-extensions >=3.7.4.3
   run_constrained:
+    # cli
+    - InquirerPy ==0.3.4
+    # inference
+    - minijinja>=1.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - minijinja >=1.0
```

**Review by skupr-anaconda — APPROVED** [review]
LGTM but a lower case for `inquirerpy` is still relevant, see https://pypi.org/project/inquirerpy/

**Review by Arishamays1 — APPROVED** [review]

**Review by Jrice1317 — APPROVED** [review]

**Pull request merged**